utils/billgeneration.js

```diff
@@ -1,20 +1,68 @@
 /* eslint-disable import/imports-first */
 import moment from 'moment';
 import dotenv from 'dotenv-safe';
+import fs from 'fs';
 
+import React from 'react';
+import ReactPDF from '@react-pdf/renderer';
 import {
   getSolarProjectById,
   getOwnerById,
   getRateScheduleById,
   createSubscriberBill,
   updateSubscriberBill,
-  getSubscriberBillsByStatementNumber
+  getSubscriberBillsByStatementNumber,
+  getSubscriberBillById
 } from '../airtable/request';
 import getEnphaseData from './enphase';
 import getLatestBill from './utilityApi';
+import BillingTemplate from './pdf/BillingTemplate';
+
+// const ROOT_URL = 'https://1f7c24d2.ngrok.io';
+const ROOT_URL = 'https://peoplepower-node.herokuapp.com';
 
 dotenv.config();
 
+const generatePdf = async (
+  subscriber,
+  solarProject,
+  subscriberBillId,
+  prevBillId
+) => {
+  // Refresh information about existing bills to get the latest calculated values
+  const subscriberBill = await getSubscriberBillById(subscriberBillId);
+  let prevBill;
+  if (prevBillId) {
+    prevBill = await getSubscriberBillById(prevBillId);
+  } else {
+    prevBill = {
+      amountDue: 0,
+      amountReceived: 0,
+      balance: 0
+    };
+  }
+  console.log(`Creating PDF for Bill# ${subscriberBill.id} ...`);
+  await ReactPDF.render(
+    <BillingTemplate
+      subscriber={subscriber}
+      solarProject={solarProject}
+      subscriberBill={subscriberBill}
+      prevBill={prevBill}
+    />,
+    `./temp/${subscriberBill.id}.pdf`
+  );
+  await updateSubscriberBill(subscriberBill.id, {
+    billPdf: [{ url: `${ROOT_URL}/${subscriberBill.id}.pdf` }]
+  });
+  console.log(
+    `Succesfully uploaded PDF at ${ROOT_URL}/${subscriberBill.id}.pdf`
+  );
+  setTimeout(() => {
+    console.log(`Deleting Temporary PDF: ${subscriberBill.id}.pdf`);
+    fs.unlinkSync(`./temp/${subscriberBill.id}.pdf`);
+  }, 20000);
```

Comment:
i guess because we're using async await here, we will only ever cal this line after we've successfully updated the subscriber bill? Do we need the timeout then in this case?

Reply:
Yeah so what I learned was that, when you upload a url to airtable, it downloads the pdf and then reuploads it to its own servers. 

So the PDF needs to exist on our servers for long enough for airtable to copy it over. I arbitrarily set it at 20 seconds before it deletes.

Reply:
wait that's so clean. wtf airtable ++ . I guess a potential issue here then, is that there might be cases when airtable doesn't actually download it in 20s, but that's an edge case we can address in the future. 20000 in Constants.js so we can change it easily in the future?

Reply:
sounds good

Reply:
I think if it it doesn't do it in 20 seconds you can just regenerate but moved to constants